pilot/pkg/xds/debug.go

```diff
@@ -293,6 +297,8 @@ func (s *DiscoveryServer) endpointz(w http.ResponseWriter, req *http.Request) {
 			all := s.Env.ServiceDiscovery.InstancesByPort(ss, p.Port, nil)
 			_, _ = fmt.Fprintf(w, "\n{\"svc\": \"%s:%s\", \"ep\": [\n", ss.Hostname, p.Name)
 			for _, svc := range all {
+				// prevent crash during serialization
```

Comment:
Maybe provide a hint as to why it would crash for future readers/maintainers? (Race on shared data?)
Are we sure `DeepCopy()` is thread safe?